packages/backend/src/studio-api-impl.ts

```diff
@@ -1,28 +1,72 @@
 import type { StudioAPI } from '@shared/src/StudioAPI';
 import type { Category } from '@shared/src/models/ICategory';
 import type { Recipe } from '@shared/src/models/IRecipe';
-import content from './ai.json';
+import defaultCatalog from './ai.json';
 import type { ApplicationManager } from './managers/applicationManager';
-import { AI_STUDIO_FOLDER } from './managers/applicationManager';
 import type { RecipeStatusRegistry } from './registries/RecipeStatusRegistry';
 import type { RecipeStatus } from '@shared/src/models/IRecipeStatus';
 import type { ModelInfo } from '@shared/src/models/IModelInfo';
 import type { TaskRegistry } from './registries/TaskRegistry';
 import type { Task } from '@shared/src/models/ITask';
-import * as path from 'node:path';
 import type { PlayGroundManager } from './playground';
 import * as podmanDesktopApi from '@podman-desktop/api';
 import type { QueryState } from '@shared/src/models/IPlaygroundQueryState';
+import type { Catalog } from '@shared/src/models/ICatalog';
+
+import * as path from 'node:path';
+import * as fs from 'node:fs';
 
 export const RECENT_CATEGORY_ID = 'recent-category';
 
 export class StudioApiImpl implements StudioAPI {
+  private catalog: Catalog;
+
   constructor(
     private applicationManager: ApplicationManager,
     private recipeStatusRegistry: RecipeStatusRegistry,
     private taskRegistry: TaskRegistry,
     private playgroundManager: PlayGroundManager,
-  ) {}
+  ) {
+    // We start with an empty catalog, for the methods to work before the catalog is loaded
+    this.catalog = {
+      categories: [],
+      models: [],
+      recipes: [],
+    };
+  }
+
+  async loadCatalog() {
+    const catalogPath = path.resolve(this.applicationManager.appUserDirectory, 'catalog.json');
+    try {
+      // TODO(feloy): watch catalog file and update catalog with new content
+      await fs.promises
+        .readFile(catalogPath, 'utf-8')
+        .then((data: string) => {
+          try {
+            const cat = JSON.parse(data) as Catalog;
+            // TODO(feloy): check version and format
+            console.log('using user catalog');
+            this.setNewCatalog(cat);
+          } catch (err: unknown) {
+            console.error('unable to parse catalog file, reverting to default catalog', err);
+            this.setNewCatalog(defaultCatalog);
+          }
+        })
+        .catch((err: unknown) => {
+          console.error('got err', err);
+          console.error('unable to read catalog file, reverting to default catalog', err);
+          this.setNewCatalog(defaultCatalog);
+        });
```

Comment:
not sure about chaining promises with the await

```ts
try {
  const data = await fs.promises.readFile(catalogPath, 'utf-8')
  const cat = JSON.parse(data) as Catalog;
  this.setCatalog(cat);
} catch (
  console.error(...)
  this.setCatalog(defaultCatalog);
}
```

I would check if the file exists before trying to read it as well.

Reply:
It's much simpler, thanks :)

Reply:
IMHO, readFile correctly and efficiently checks the existence of the file, we don't have to check it before (or we would have to handle another error case)

Reply:
it's mainly b/c usually the absence of the file is not an error so you want to read it if you know user provided a custom file

and getting an exception b/c the file is not there or getting an exception b/c we're unable to read the file is not the same while here we handle it

it looks like we'll get a log trace

```
unable to read catalog file, reverting to default catalog
```

even if we don't have customized the file

Reply:
I agree about the error log trace. Let's add the check and be silent if the file doesn't exist